fdbclient/S3BlobStore.actor.cpp

```diff
@@ -909,30 +920,44 @@ ACTOR Future<Reference<HTTP::Response>> doRequest_impl(Reference<S3BlobStoreEndp
 
 			remoteAddress = rconn.conn->getPeerAddress();
 			wait(bstore->requestRate->getAllowance(1));
-			Reference<HTTP::Response> _r = wait(timeoutError(HTTP::doRequest(rconn.conn,
-			                                                                 verb,
-			                                                                 canonicalURI,
-			                                                                 headers,
-			                                                                 &contentCopy,
-			                                                                 contentLen,
-			                                                                 bstore->sendRate,
-			                                                                 &bstore->s_stats.bytes_sent,
-			                                                                 bstore->recvRate),
-			                                                 requestTimeout));
+
+			Future<Reference<HTTP::Response>> reqF = HTTP::doRequest(rconn.conn,
+			                                                         verb,
+			                                                         canonicalURI,
+			                                                         headers,
+			                                                         &contentCopy,
+			                                                         contentLen,
+			                                                         bstore->sendRate,
+			                                                         &bstore->s_stats.bytes_sent,
+			                                                         bstore->recvRate);
+
+			// if we reused a connection from the pool, and immediately got an error, retry immediately discarding the
+			// connection
+			if (reqF.isReady() && reusingConn) {
```

Comment:
This should probably be `isError()` for clarity to match the comment, even though in practice it would be the same behavior (unless there were a mock doRequest() implementation that could return HTTP 500's etc instantly).